src/chunks/norms.rs

```diff
@@ -24,11 +18,29 @@ pub trait Norms {
 /// The unnormalized embedding can be reconstructed by multiplying the
 /// normalized embedding by its orginal l2 norm.
 #[derive(Clone, Debug)]
-pub struct NdNorms(pub Array1<f32>);
+pub struct NdNorms {
+    norms: Array1<f32>,
+}
+
+impl NdNorms {
+    /// Construct new `NdNorms`.
+    pub fn new<T>(norms: T) -> Self
+    where
+        T: Into<Array1<f32>>,
+    {
+        NdNorms {
+            norms: norms.into(),
+        }
+    }
+
+    /// Return the norm for the word at the given index.
```

Comment:
I am wondering whether it makes sense to implement `Deref` here (`DerefMut` is not necessary here). Then we wouldn't need any custom methods. Of course, this would require that the internal representation is always `Array1`, which we may not want to commit to?

Reply:
That was the intuition behind this, we might be able to provide `ArrayView1` with different backing storages. On the other hand, our constructor asks for `Into<Array1>` so we might aswell commit to `Array1`.